File: C++/A/821A-okabeandfuturegadgetlab.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int dimension;
    vector<vector<int>> lab;

public:
    void input()
    {
        cin >> dimension;
        for (int i = 0; i < dimension; ++i)
        {
            vector<int> row;
            for (int j = 0; j < dimension; j++)
            {
                int x;
                cin >> x;
                row.push_back(x);
            }
            lab.push_back(row);
        }
    }
    bool lab_validator()
    {
        for (int i = 0; i < dimension; i++)
        {
            for (int j = 0; j < dimension; j++)
            {
                if (lab[i][j] == 1)
                    continue;
                bool pass = false;
                for (int r = 0; r < dimension; r++)
                {
                    if (r == i)
                        continue;
                    for (int c = 0; c < dimension; c++)
                    {
                        if (c == j)
                            continue;
                        int sum = lab[r][j] + lab[i][c];
                        if (sum == lab[i][j])
                        {
                            pass = true;
                            break;
                        }
                    }
                }
                if (!pass)
                    return false;
            }
        }
        return true;
    }
};

int main()
{
    Solver solution;
    solution.input();
    cout << ((solution.lab_validator()) ? "Yes" : "No");
}

File: Python/A/821A-okabeandfuturegadgetlab.py
class Solver:
    def __init__(self):
        self.dimension = 0
        self.lab = []

    def input_data(self):
        self.dimension = int(input())
        for _ in range(self.dimension):
            row = list(map(int, input().split()))
            self.lab.append(row)

    def lab_validator(self):
        for i in range(self.dimension):
            for j in range(self.dimension):
                if self.lab[i][j] == 1:
                    continue
                pass_flag = False
                for r in range(self.dimension):
                    if r == i:
                        continue
                    for c in range(self.dimension):
                        if c == j:
                            continue
                        _sum = self.lab[r][j] + self.lab[i][c]
                        if _sum == self.lab[i][j]:
                            pass_flag = True
                            break
                if not pass_flag:
                    return False
        return True


solution = Solver()
solution.input_data()
print("Yes" if solution.lab_validator() else "No")
